### PR TITLE
Fixed the policy bug when managed hub rejoin with a different name

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>

Manager Log:
```bash
2023/09/08 08:17:37 /workspace/manager/pkg/statussyncer/syncers/local_spec_policies_syncer.go:110 pq: duplicate key value violates unique constraint "policies_pkey"
[2.836ms] [rows:0] INSERT INTO "local_spec"."policies" ("leaf_hub_name","payload") VALUES ('regional-28','{"kind":"Policy","apiVersion":"policy.open-cluster-management.io/v1","metadata":{"name":"backup-restore-enabled","namespace":"open-cluster-management-backup","uid":"6f3be17e-6882-4989-8dc2-c7b7889925dc","resourceVersion":"261875","creationTimestamp":"2023-09-05T07:08:37Z","labels":{"app":"cluster-backup-chart","app.kubernetes.io/instance":"cluster-backup-chart","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"cluster-backup-chart","chart":"cluster-backup-chart","component":"policy","helm.sh/chart":"cluster-backup-chart","heritage":"Helm","installer.name":"multiclusterhub","installer.namespace":"open-cluster-management","release":"cluster-backup-chart","velero.io/exclude-from-backup":"true"},"annotations":{"policy.open-cluster-management.io/categories":"PR.IP Information Protection Processes and Procedures","policy.open-cluster-management.io/controls":"PR.IP-4 Backups of information are conducted maintained and tested","policy.open-cluster-management.io/source":"system","policy.open-cluster-management.io/standards":"NIST-CSF"}},"spec":{"disabled":false,"remediationAction":"inform","policy-templates":[{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"acm-backup-pod-running"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"v1","kind":"Pod","metadata":{"labels":{"app":"cluster-backup-chart"},"namespace":"open-cluster-management-backup"},"status":{"phase":"Running"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"oadp-pod-running"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"v1","kind":"Pod","metadata":{"annotations":{"repository":"https://github.com/openshift/oadp-operator"},"namespace":"open-cluster-management-backup"},"status":{"phase":"Running"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"velero-pod-running"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"v1","kind":"Pod","metadata":{"labels":{"app.kubernetes.io/name":"velero"},"namespace":"open-cluster-management-backup"},"status":{"phase":"Running"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"data-protection-application-available"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"oadp.openshift.io/v1alpha1","kind":"DataProtectionApplication","metadata":{"namespace":"open-cluster-management-backup"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"backup-storage-location-available"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"velero.io/v1","kind":"BackupStorageLocation","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":"Available"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"backup-schedule-cron-enabled"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"velero.io/v1","kind":"Backup","metadata":{"labels":{"velero.io/schedule-name":"acm-validation-policy-schedule"},"namespace":"open-cluster-management-backup"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"acm-managed-clusters-schedule-backups-available"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"velero.io/v1","kind":"Backup","metadata":{"labels":{"velero.io/schedule-name":"acm-managed-clusters-schedule"},"namespace":"open-cluster-management-backup"},"status":{"phase":"Completed"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"acm-resources-schedule-backups-available"},"spec":{"object-templates":[{"complianceType":"musthave","objectDefinition":{"apiVersion":"velero.io/v1","kind":"Backup","metadata":{"labels":{"velero.io/schedule-name":"acm-resources-schedule"},"namespace":"open-cluster-management-backup"},"status":{"phase":"Completed"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"acm-backup-in-progress-report"},"spec":{"object-templates":[{"complianceType":"mustnothave","objectDefinition":{"apiVersion":"velero.io/v1","kind":"Backup","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":"InProgress"}}}],"remediationAction":"inform","severity":"medium"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"acm-backup-clusters-collision-report"},"spec":{"object-templates":[{"complianceType":"mustnothave","objectDefinition":{"apiVersion":"cluster.open-cluster-management.io/v1beta1","kind":"BackupSchedule","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":"BackupCollision"}}}],"remediationAction":"inform","severity":"high"}}},{"objectDefinition":{"apiVersion":"policy.open-cluster-management.io/v1","kind":"ConfigurationPolicy","metadata":{"name":"acm-backup-phase-validation"},"spec":{"object-templates":[{"complianceType":"mustnothave","objectDefinition":{"apiVersion":"cluster.open-cluster-management.io/v1beta1","kind":"BackupSchedule","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":"FailedValidation"}}},{"complianceType":"mustnothave","objectDefinition":{"apiVersion":"cluster.open-cluster-management.io/v1beta1","kind":"BackupSchedule","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":"Failed"}}},{"complianceType":"mustnothave","objectDefinition":{"apiVersion":"cluster.open-cluster-management.io/v1beta1","kind":"BackupSchedule","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":""}}},{"complianceType":"mustnothave","objectDefinition":{"apiVersion":"cluster.open-cluster-management.io/v1beta1","kind":"Restore","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":""}}},{"complianceType":"mustnothave","objectDefinition":{"apiVersion":"cluster.open-cluster-management.io/v1beta1","kind":"Restore","metadata":{"namespace":"open-cluster-management-backup"},"status":{"phase":"Error"}}}],"remediationAction":"inform","severity":"high"}}}]},"status":{}}') RETURNING "policy_id","policy_name","policy_standard","policy_category","policy_control","created_at","updated_at","deleted_at"
```

### CODE REVIEW

File: manager/pkg/statussyncer/syncers/local_spec_policies_syncer.go [hunk 1]
```diff
@@ -107,6 +107,10 @@ func (syncer *localSpecPoliciesSyncer) handleLocalObjectsBundle(ctx context.Cont
 
 			// if the row doesn't exist in db then add it.
 			if !objInDB {
+				syncer.log.Info("local policy created", "leafHubName", leafHubName, "policyID", uid)
```
Comment (clyang82) [inline manager/pkg/statussyncer/syncers/local_spec_policies_syncer.go:110]: from the comments
> // if the row doesn't exist in db then add it.

That means the row does not exist. why need deleting this row?

Comment (yanmxa) [reply manager/pkg/statussyncer/syncers/local_spec_policies_syncer.go:110]: It just means this [row doesn't exist on this managed hub](https://github.com/stolostron/multicluster-global-hub/pull/626/files#diff-ddd5c421e780d61c347a155fd2d30b1cda6ad4e19c23c4e5fdad939b1e2614c1R170). This aims to address the issue when the managed hub name is changed.

**Review by clyang82 — APPROVED** [review]

**Pull request merged**